lib/gui/app/models/store.ts

```diff
@@ -491,8 +494,8 @@ function storeReducer(
 			);
 		}
 
-		case Actions.DESELECT_IMAGE: {
-			return state.deleteIn(['selection', 'image']);
+		case Actions.DESELECT_SOURCE: {
+			return state.setIn(['selection', 'image'], {});
```

Comment:
Deleting it was a better option. Now we need to check for empty objects.

Reply:
So what do you prefer, having the image undefined or empty when it's not selected?

Reply:
`undefined`